crypto/bio/bio_test.cc

```diff
@@ -154,6 +154,64 @@ TEST(BIOTest, Printf) {
   }
 }
 
+TEST(BIOTest, TextFile) {
+#if defined(OPENSSL_ANDROID)
+  // On Android, when running from an APK, |tmpfile| does not work. See
+  // b/36991167#comment8.
+  GTEST_SKIP();
+#endif
+
+
+  // unique_ptr will automatically call fclose on the file descriptior when the
+  // variable goes out of scope, so we need to specify BIO_NOCLOSE close flags
+  // to avoid a double-free condition.
+  using TempFILE = std::unique_ptr<FILE, decltype(&fclose)>;
```

Comment:
I don't see any tests here that utilize `BIO_CLOSE` to verify that it does close the file when specified.

Reply:
Huh, [searching around for `BIO_CLOSE`](https://github.com/search?q=repo%3Aaws%2Faws-lc+BIO_CLOSE&type=code) it looks like we don't have any direct test coverage on that flag being set explicitly, even though it's [documented as the default](https://github.com/aws/aws-lc/blob/aa90e509f2e940916fbe9fdd469a4c90c51824f6/include/openssl/bio.h#L392). I'll add a test case for this.

Reply:
Calling `BIO_CLOSE` the "default" could be misleading. If you invoke `BIO_new_fp(text_bio_file.get(), BIO_FP_TEXT);`, there's no way for [`BIO_new_fp`](https://github.com/aws/aws-lc/blob/aa90e509f2e940916fbe9fdd469a4c90c51824f6/crypto/bio/file.c#L118) to know that you might expect the "default" to provide the `BIO_CLOSE` flag behavior.

Reply:
Hm, fair point. I'll see if I can reword this.